packages/web-app-files/tests/unit/components/SideBar/Links/FileLinks.spec.js

```diff
@@ -0,0 +1,240 @@
+import { mount, shallowMount, createLocalVue } from '@vue/test-utils'
+import FileLinks from '@files/src/components/SideBar/Links/FileLinks.vue'
+
+const localVue = createLocalVue()
+
+localVue.use(Vuex)
+localVue.use(DesignSystem)
+localVue.use(GetTextPlugin, {
+  translations: 'does-not-matter.json',
+  silent: true
+})
+
+const defaultLinksList = [
+  {
+    id: '1',
+    indirect: false,
+    name: 'public link 1',
+    url: 'some-link-1',
+    path: '/file-1.txt'
+  },
+  {
+    id: '2',
+    indirect: true,
+    name: 'public link 2',
+    url: 'some-link-2',
+    path: '/file-2.txt'
+  }
+]
+
+const mapActions = {
+  loadSharesTree: jest.fn(),
+  loadCurrentFileOutgoingShares: jest.fn()
+}
+
+const mapMutations = {
+  TRIGGER_PUBLIC_LINK_CREATE: jest.fn()
+}
+
+describe('FileLinks', () => {
+  const selectors = {
+    linkAddButton: '#files-file-link-add',
+    noResharePermissions: 'p[data-testid="files-links-no-reshare-permissions-message"]',
+    linkNoResults: '#oc-file-links-no-results',
+    listItemStub: 'list-item-stub'
+  }
+
+  function createStore({
+    links = defaultLinksList,
+    highlightedFile = {
+      path: '/lorem.txt',
+      type: 'file',
+      canShare: jest.fn(() => true)
+    },
+    currentFileOutgoingSharesLoading = false,
+    sharesTreeLoading = false,
+    expireDate = {
+      enabled: true,
+      days: 1,
+      enforced: '1'
+    },
+    privateLinks = false
+  } = {}) {
+    return new Vuex.Store({
+      getters: {
+        capabilities: jest.fn(() => {
+          return {
+            files: {
+              privateLinks: privateLinks
+            },
+            files_sharing: {
+              public: {
+                defaultPublicLinkShareName: 'public link name default',
+                expire_date: expireDate
+              }
+            }
+          }
+        })
+      },
+      modules: {
+        Files: {
+          namespaced: true,
+          state: {
+            sharesTree: {}
+          },
+          getters: {
+            highlightedFile: function() {
+              return highlightedFile
+            },
+            currentFileOutgoingLinks: function() {
+              return links
+            },
+            currentFileOutgoingSharesLoading: jest.fn(() => currentFileOutgoingSharesLoading),
+            sharesTreeLoading: jest.fn(() => sharesTreeLoading)
+          },
+          actions: mapActions,
+          mutations: mapMutations
+        }
+      }
+    })
+  }
+
+  function getShallowWrapper(store) {
+    return shallowMount(FileLinks, {
+      localVue,
+      store: store,
+      stubs: stubs,
+      provide: {
+        changeView: jest.fn()
+      }
+    })
+  }
+
+  function getMountedWrapper(store) {
+    return mount(FileLinks, {
+      localVue,
+      store: store,
+      stubs: {
+        'link-edit': true,
+        'list-item': true
+      },
+      provide: {
+        changeView: jest.fn()
+      }
+    })
+  }
+
+  describe('current view', () => {
```

Comment:
can we find a more descriptive string here? "current view" might apply to a lot of places